File: 30-06-2024_Delete_node_in_Doubly_Linked_List.java
class Solution {
    public Node deleteNode(Node head, int x) {
        Node del = head;
        x = x - 1;
        while (x-- > 0) del = del.next;

        /* base case */
        // if head_ref or del is null, return null.
        if (head == null || del == null) return null;

        /* If Node to be deleted is head Node */
        // if del is equal to head_ref, update head_ref to next node.
        if (head == del) head = del.next;

        /* Change next only if Node to be deleted is NOT the last Node */
        // if del next is not null, update previous node of del next to del previous.
        if (del.next != null) del.next.prev = del.prev;

        /* Change prev only if Node to be deleted is NOT the first Node */
        // if del previous is not null, update next node of del previous to del next.
        if (del.prev != null) del.prev.next = del.next;

        // return updated head_ref.
        return head;
    }
}

File: 30-06-2024_Delete_node_in_Doubly_Linked_List.py
class Solution:

    def delete_node(self, head, x):
        del_node = head
        x -= 1
        while x > 0:
            del_node = del_node.next
            x -= 1

        # base case
        if head is None or del_node is None:
            return None

        # If Node to be deleted is head Node
        if head == del_node:
            head = del_node.next

        # Change next only if Node to be deleted is NOT the last Node
        if del_node.next is not None:
            del_node.next.prev = del_node.prev

        # Change prev only if Node to be deleted is NOT the first Node
        if del_node.prev is not None:
            del_node.prev.next = del_node.next

        # return updated head
        return head
